src/compas_fab/robots/robot.py

```diff
@@ -47,15 +48,43 @@ class Robot(object):
         Dictionary mapping planning groups to the tool currently attached to them, if any.
     """
 
-    def __init__(self, model, artist=None, semantics=None, client=None):
+    def __init__(self, model=None, artist=None, semantics=None, client=None):
+        super(Robot, self).__init__()
+        # These attributes have to be initiated first,
+        # because they are used in the setters of the other attributes
         self._scale_factor = 1.0
-        self.model = model
         self._attached_tools = {}  # { planning_group_name: robots.tool.Tool }
+        self._current_ik = {"request_id": None, "solutions": None}
+
+        self.model = model
         self.artist = artist
         self.semantics = semantics
         self.client = client
         self.attributes = {}
-        self._current_ik = {"request_id": None, "solutions": None}
+
+    @property
+    def data(self):
+        data = {
+            "_scale_factor": self._scale_factor,
+            "_attached_tools": self._attached_tools,
+            "_current_ik": self._current_ik,
```

Comment:
why the `_`s?

Reply:
I didn't invent this particular variable, and it was there all along, I just serialized it. I believe the intent of the existing code is that the model is not supposed to keep track of extrinsic or modifiable properties of the robot (in this case, the `current_ik` ). Hence the private `_` notation. I should probably have another PR to address the intrinsic vs extrinsic state of the robot problem.

Reply:
I would remove the `_` in the keys.

Reply:
also, `current_ik` should not be serialized